django/db/backends/postgresql/base.py

```diff
@@ -598,6 +844,36 @@ def copy(self, statement):
             with self.debug_sql(statement):
                 return self.cursor.copy(statement)
 
+    class AsyncServerBindingCursor(CursorMixin, Database.AsyncClientCursor):
+        pass
+
+    class AsyncCursor(CursorMixin, Database.AsyncClientCursor):
+        pass
+
+    class AsyncServerSideCursor(
+        CursorMixin,
+        Database.client_cursor.ClientCursorMixin,
+        Database.AsyncServerCursor,
+    ):
+        """
+        psycopg >= 3 forces the usage of server-side bindings when using named
+        cursors but the ORM doesn't yet support the systematic generation of
+        prepareable SQL (#20516).
+
+        ClientCursorMixin forces the usage of client-side bindings while
+        ServerCursor implements the logic required to declare and scroll
+        through named cursors.
+
+        Mixing ClientCursorMixin in wouldn't be necessary if Cursor allowed to
```

Comment:
Perhaps out of scope for this PR, but I don't think it's clear where this `Cursor` comes from. I *think* it's defined in `psycopg3`, but I'm not confident in that.

This should also be clarified in the `ServerSideCursor` docstring above too.

Reply:
anything under `Database.` comes from `psycopg2` or `psycopg3`.

In this specific case, `ClientCursonMixin` and `AsyncClientCursor` come from `psycopg3`

`ServerSideCursor` Is our custom class that derives from postgres' `ClientCursorMixin`, `AsyncServerCursor`/`ServerCursor`